source/core/slang-string.h

```diff
@@ -135,10 +135,14 @@ namespace Slang
     {
     public:
         UnownedStringSlice()
-            : beginData(0)
+            : beginData(nullptr)
             , endData(0)
         {}
 
+        explicit UnownedStringSlice(char const* a):
```

Comment:
Just to be clear, I left out this constructor intentionally. It's a mouthful, but the `UnownedTerminatedStringSlice` type extends `UnownedStringSlice` with the invariant that the byte one after `end` is zero, and it includes the implicit conversion from `const char*`, so you can always use the `UnownedTerminatedStringSlice` constructor when you want the behavior of the explicit constructor you added here, which makes your intention explicit ("I know this `char*` represents a terminated string").

Another option I've seen people do to handle literal strings is a bit of template hackery where you make the constructor a template on `size_t N`, and then take a reference to an array of size `[N]` (because a reference to an array won't decay to a pointer...). I'm not advocating for that because template trickery can get gross, but it could also work.

I'm fine with leaving this in as an explicit constructor for now, since the design intent wasn't documented before, and there is a reasonable discussion to be had about what API makes sense for these basic types.

Reply:
I think you are right that making the constructor `explicit` eliminates most of the problems around accidental conversions. I expect that initializing from fixed-size arrays of `char` that are *not* string literals is unlikely to occur in this codebase, so I'm less worried about that case coming up.